pkg/kubectl/cmd/config/unset_test.go

```diff
@@ -117,3 +117,42 @@ func (test unsetConfigTest) run(t *testing.T) {
 		}
 	}
 }
+
+func TestUnsetUnexistConfig(t *testing.T) {
+	conf := clientcmdapi.Config{
+		Kind:       "Config",
+		APIVersion: "v1",
+		Clusters: map[string]*clientcmdapi.Cluster{
+			"minikube":   {Server: "https://192.168.99.100:8443"},
+			"my-cluster": {Server: "https://192.168.0.1:3434"},
+		},
+		Contexts: map[string]*clientcmdapi.Context{
+			"minikube":   {AuthInfo: "minikube", Cluster: "minikube"},
+			"my-cluster": {AuthInfo: "mu-cluster", Cluster: "my-cluster"},
+		},
+		CurrentContext: "minikube",
+	}
+	fakeKubeFile, err := ioutil.TempFile(os.TempDir(), "")
```

Comment:
I really wish you had re-used/re-worked `run()` to not do this duplication. It looks like effort was put in to make things re-usable but then we didn't re-use them. Admittedly you'll have to make changes so there is an 'expectedErr' inside `unsetConfigTest`.

Other than that nit, which you you tell me you'd rather not do, it lgtm.

Reply:
original intention i want to distinguish success and fail cases. but seems other cases put them together. i'd more like follow your advise and put them together. and seems i add `expectErr`   to ```unsetConfigTest```. this ```run()``` will became more reuseable. i like do this:-) thank you very much